pkg/composition/placement-request.go

```diff
@@ -0,0 +1,87 @@
+package composition
+
+import (
+	"fmt"
+	"slices"
+
+	"kombiner/pkg/apis/v1alpha1"
+)
+
+// PlacementRequest purpose is to provide auxiliar methods to an existing
+// v1alpha1.PlacementRequest.
+type PlacementRequest struct {
+	*v1alpha1.PlacementRequest
+}
+
+// SetBindingFailure helps when adding a bind failure condition to the
+// PlacementRequest.
+func (pr *PlacementRequest) SetBindingFailure(bind v1alpha1.Binding, reason, msg string) {
+	pr.Status.Bindings = append(
+		pr.Status.Bindings,
+		v1alpha1.PlacementRequestBindingResult{
+			Binding: bind,
+			Result:  v1alpha1.PlacementRequestResultFailure,
+			Reason:  reason,
+			Message: msg,
+		},
+	)
+}
+
+// SetBindingSuccess helps when setting a successful binding condition for a
+// given v1alpha1.Binding.
+func (pr *PlacementRequest) SetBindingSuccess(bind v1alpha1.Binding) {
+	pr.Status.Bindings = append(
+		pr.Status.Bindings,
```

Comment:
It might work better to set a binding success at a specific index so the order of invocation does not make a difference to avoid any unforeseen side effects.

Reply:
Yes, this makes total sense.